.github/workflows/reusable_dockerfile_pipeline.yml

```diff
@@ -35,25 +36,57 @@ jobs:
       - name: Add vars to ENV
         id: setting_env
         run: |
-          echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
-          echo "short_sha=`echo ${GITHUB_SHA} | cut -c1-8`" >> "$GITHUB_OUTPUT"
-          # yamllint disable
-          echo "IMAGE_NAME=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
-          echo "image_name=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> "$GITHUB_OUTPUT"
-          # here we validate if we have specified a different package name in
-          # the inputs, if so, we change the package to it.
-          if [[ ${{ inputs.packageName }} != ${{ github.repository}} ]];then
+          # Extract both the repository owner and repository name
+          # According to docs, github.repository is in the format of owner/repo, not just repo
+          # https://docs.github.com/en/actions/learn-github-actions/contexts#github-context
+            REPO_OWNER=$(echo "${{ github.repository }}" | awk -F'/' '{print $1}' | tr '[:upper:]' '[:lower:]')
+            REPO_NAME=$(echo "${{ github.repository }}" | awk -F'/' '{print $2}' | tr '[:upper:]' '[:lower:]')
+
+            echo "Repository Owner: $REPO_OWNER"
+            echo "Repository Name: $REPO_NAME"
+
+            echo "REPO_OWNER=$REPO_OWNER" >> $GITHUB_ENV
+            echo "repo_owner=$REPO_OWNER" >> "$GITHUB_OUTPUT"
+
+            echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
+            echo "short_sha=`echo ${GITHUB_SHA} | cut -c1-8`" >> "$GITHUB_OUTPUT"
+            # yamllint disable
+
+            echo "IMAGE_NAME=$REPO_NAME" >> $GITHUB_ENV
+            echo "image_name=$REPO_NAME" >> "$GITHUB_OUTPUT"
+
+            # here we validate if we have specified a different package name in
+            # the inputs, if so, we change the package to it.
+            if [[ ${{ inputs.packageName }} != ${{ github.repository}} ]];then
             # validate the input package name characters
-            if [[ ! "${{ inputs.packageName }}" =~ ^[A-Za-z0-9\-]+$ ]]; then
-              echo "------------------------------------------------------------"
-              echo "ERROR: Package name not valid! => [ ${{ inputs.packageName }} ]"
-              echo "ONLY can use: A-Za-z0-9\-"
-              echo "------------------------------------------------------------"
-              exit 1
+              if [[ ! "${{ inputs.packageName }}" =~ ^[A-Za-z0-9\-]+$ ]]; then
+                echo "------------------------------------------------------------"
+                echo "ERROR: Package name not valid! => [ ${{ inputs.packageName }} ]"
+                echo "ONLY can use: A-Za-z0-9\-"
+                echo "------------------------------------------------------------"
+                exit 1
+              fi
+
+              # Set the default package name to the repository name
+              PACKAGE_NAME=$REPO_NAME
```

Comment:
also quick gut check here. 

We were setting `PACKAGE_NAME=$REPO_NAME` but in the original code we were checking `inputs.packageName != github.repository` 

Should we be setting `PACKAGE_NAME=github.repository` instead?

Reply:
setting `PACKAGE_NAME=github.repository` is not going to work, as the latter is a combination of repo's owner and repo name itself. you can see at line 40-41 the repo name is taken from github.repository 2nd part of the string.

Reply:
> Was the original check just invalid?

Yes, bc: 

Here is the original checker 
```sh
 echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
          echo "short_sha=`echo ${GITHUB_SHA} | cut -c1-8`" >> "$GITHUB_OUTPUT"
          # yamllint disable
          echo "IMAGE_NAME=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
          echo "image_name=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> "$GITHUB_OUTPUT"
          # here we validate if we have specified a different package name in
          # the inputs, if so, we change the package to it.
          if [[ ${{ inputs.packageName }} != ${{ github.repository}} ]];then
            # validate the input package name characters
            if [[ ! "${{ inputs.packageName }}" =~ ^[A-Za-z0-9\-]+$ ]]; then
              echo "------------------------------------------------------------"
              echo "ERROR: Package name not valid! => [ ${{ inputs.packageName }} ]"
              echo "ONLY can use: A-Za-z0-9\-"
              echo "------------------------------------------------------------"
              exit 1
            fi
            echo "IMAGE_NAME=$(echo ${{ github.repository_owner }}/${{ inputs.packageName }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
            echo "image_name=$(echo ${{ github.repository_owner }}/${{ inputs.packageName }} | tr '[:upper:]' '[:lower:]')" >> "$GITHUB_OUTPUT"
          fi
          # yamllint enable
```

Here is the usage of the inputs.packageName in app's txsim build stage
```sh yaml
  docker-txsim-build:
    permissions:
      contents: write
      packages: write
    uses: celestiaorg/.github/.github/workflows/reusable_dockerfile_pipeline.yml@v0.2.8
    with:
      dockerfile: docker/Dockerfile_txsim
      packageName: txsim
```

It was just checking that if the package name is not the canonical repository name of `celestiaorg/celestia-app`. 
i.e. 
```sh 
if 'txsim' != 'celestiaorg/celestia-app'
```
The github.repository name will always provide a owner+repo_name. 


This line from the original script pretty much sums it all how we add the namespace
```sh
echo "IMAGE_NAME=$(echo ${{ github.repository_owner }}/${{ inputs.packageName }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
```

> We might need to add it as a stand alone check below to warn about that edge case.

Why? Can you elaborate here?